src/component/NodeEditor/NodeEditor.test.tsx

```diff
@@ -224,4 +222,117 @@ describe('NodeEditor >', () => {
             ]);
         });
     });
+
+    describe('instance methods >', () => {
+        describe('updateSubflowRouter >', () => {
+            it("it should call 'updateRouter' with the existing exit, cases if node exists and is of type 'subflow'", () => {
+                const updateRouterSpy = jest.spyOn(NodeEditor.prototype, 'updateRouter');
+                const updateSubflowRouterSpy = jest.spyOn(
+                    NodeEditor.prototype,
+                    'updateSubflowRouter'
+                );
+
+                const wrapper = mount(
+                    <NodeEditor
+                        {...{
+                            ...nodeEditorProps,
+                            node: subflowRouterNode,
+                            action: startFlowAction
+                        }}
+                    />,
+                    {
+                        context: { endpoints },
+                        childContextTypes: {
+                            endpoints: endpointsPT
+                        }
+                    }
+                );
+
+                wrapper.instance().submit();
+
+                expect(updateSubflowRouterSpy).toHaveBeenCalled();
+                expect(updateRouterSpy).toHaveBeenCalledWith(
+                    expect.objectContaining({
+                        router: expect.objectContaining({
+                            cases: subflowRouterNode.router.cases
+                        }),
+                        exits: subflowRouterNode.exits
+                    }),
+                    'subflow',
+                    startFlowAction
+                );
+
+                updateSubflowRouterSpy.mockRestore();
+                updateRouterSpy.mockRestore();
+            });
+        });
+
+        it("it should call 'updateRouter' with the new exit, cases if node exists but is not of type 'subflow'", () => {
+            const updateRouterSpy = jest.spyOn(NodeEditor.prototype, 'updateRouter');
+            const updateSubflowRouterSpy = jest.spyOn(NodeEditor.prototype, 'updateSubflowRouter');
+
+            const wrapper = mount(
+                <NodeEditor {...{ ...nodeEditorProps, action: startFlowAction }} />,
+                {
+                    context: { endpoints },
+                    childContextTypes: {
+                        endpoints: endpointsPT
+                    }
+                }
+            );
+
+            wrapper.setState({ config: getTypeConfig('start_flow') });
+
+            expect(wrapper.state('config')).toEqual(
+                expect.objectContaining({
+                    type: 'start_flow'
+                })
+            );
+
+            expect(wrapper.find('SubflowRouter').prop('updateRouter')).toEqual(
+                wrapper.instance().updateSubflowRouter
+            );
+
+            wrapper.instance().submit();
+
+            expect(updateSubflowRouterSpy).toHaveBeenCalled();
+            expect(updateRouterSpy).toHaveBeenCalledWith(
```

Comment:
assumption here is if `ComponentMap.getDetails` doesn't return a `ComponentDetails` interface of type `subflow` but `NodeEditor`'s `updateRouter` method is called with an object containing two exits and cases, the right thing happened.